component/src/main/java/org/wso2/extension/siddhi/io/feed/sink/FeedSink.java

```diff
@@ -0,0 +1,268 @@
+                                " define stream outputStream(content string, title string);",
+                        description = " This example shows how to create Atom entry on existing atom document. The " +
+                                "variables of stream are the standard element of a atom entry "
+                )
+        }
+)
+
+public class FeedSink extends Sink {
+
+    private URL url;
+    private Abdera abdera;
+    private AbderaClient abderaClient;
+    private int httpResponse;
+    private String atomFunc;
+    private StreamDefinition streamDefinition;
+    private OptionHolder optionHolder;
+
+    @Override
+    public Class[] getSupportedInputEventClasses() {
+            return new Class[]{Map.class};
+    }
+
+    @Override
+    public String[] getSupportedDynamicOptions() {
+            return new String[0];
+    }
+
+    @Override
+    protected void init(StreamDefinition streamDefinition, OptionHolder optionHolder, ConfigReader configReader,
+            SiddhiAppContext siddhiAppContext) {
+        try {
+            this.url = new URL(optionHolder.validateAndGetStaticValue(Constants.URL));
+        } catch (MalformedURLException e) {
+            throw new SiddhiAppValidationException("Url Syntax Error in " + streamDefinition.getId() + ". Given value "
+                    + optionHolder.validateAndGetStaticValue(Constants.URL));
+        }
+        this.optionHolder = optionHolder;
+        this.streamDefinition = streamDefinition;
+        this.abdera = new Abdera();
+        this.abderaClient = new AbderaClient(abdera);
+        this.httpResponse = Integer.parseInt(optionHolder.validateAndGetStaticValue(Constants.HTTP_RESPONSE_CODE,
+                Constants.HTTP_CREATED));
+        this.atomFunc = validateAtom(optionHolder.validateAndGetStaticValue(Constants.ATOM_FUNC,
+                Constants.FEED_CREATE));
+
+        int timeout = validateTimeOut();
+        this.abderaClient.setConnectionTimeout(timeout);
+        String userName = optionHolder.validateAndGetStaticValue(Constants.USERNAME, Constants.CREDENTIALS);
+        String password = optionHolder.validateAndGetStaticValue(Constants.PASSWORD, Constants.CREDENTIALS);
+        if (!optionHolder.validateAndGetStaticValue(Constants.USERNAME, Constants.CREDENTIALS)
+                .equals(Constants.CREDENTIALS)
+                || !optionHolder.validateAndGetStaticValue(Constants.PASSWORD, Constants.CREDENTIALS)
+                        .equals(Constants.CREDENTIALS)) {
+            AbderaClient.registerTrustManager();
+            try {
+                abderaClient.addCredentials(String.valueOf(url), Constants.REALM, Constants.AUTH_SCHEME,
+                        new UsernamePasswordCredentials(userName, password));
+            } catch (URISyntaxException e) {
+                // nothing, url syntax exception doesn't throw because url string getting form a URL object
+            }
+        }
+    }
+
+    // atom function validation
```

Comment:
Use doc comments instead of using line comments while adding comments to methods. Check all places
eg:` /** documentation */`